src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -119,59 +125,80 @@
      * Clone, analyze and generate the report for repositories in {@code repoLocationMap}.
      * Performs analysis and report generation of each repository in parallel with the cloning of the next repository.
      */
-    private static void cloneAndAnalyzeRepos(
-            Map<RepoLocation, List<RepoConfiguration>> repoLocationMap, String outputPath) throws IOException {
+    private static void cloneAndAnalyzeRepos(List<RepoConfiguration> configs, String outputPath) {
+        Map<RepoLocation, List<RepoConfiguration>> repoLocationMap = groupConfigsByRepoLocation(configs);
         RepoCloner repoCloner = new RepoCloner();
         RepoLocation clonedRepoLocation = null;
 
-        for (RepoLocation location : repoLocationMap.keySet()) {
-            repoCloner.clone(outputPath, repoLocationMap.get(location).get(0));
+        List<RepoLocation> repoLocationList = new ArrayList<>(repoLocationMap.keySet());
+        Set<RepoLocation> successfullyClonedRepos = new HashSet<>();
+
+        // clones another repo in parallel while analyzing a cloned repo (if any)
+        for (RepoLocation location : repoLocationList) {
```

Comment:
A little change here, lets turn this into a `for` loop instead of using a `for each` loop and let the counter start from 1 and ends at n + 1. 

Main idea here is to not have an additional `if (clonedRepoLocation != null) {` outside of the loop, so that we can save some duplicate codes as well as remove the failed configs on the go instead of making use of a `successfullyClonedRepos`.